src/main/scala/util/ClockGate.scala

```diff
@@ -23,6 +35,8 @@ object ClockGate {
       name: Option[String] = None)(implicit p: Parameters): Clock = {
     val cg = Module(p(ClockGateImpl)())
     name.foreach(cg.suggestName(_))
+    p(ClockGateModelFile).map(cg.addVerilogResource(_))
```

Comment:
I'm not so sure if this correct. Have you tried this? using `addPath` and `addResource` out of `BlackBox` Module is a little anti-pattern. IMHO, you can directly use the `CDE` parameter there.

Reply:
Yes, following is from the annotation file, the `target` field is correctly set to EICG_wrapper:
```
  {
    "class":"firrtl.transforms.BlackBoxPathAnno",
    "target":"TestHarness.EICG_wrapper",
    "path":"/.../rocket-chip/src/main/resources/vsrc/EICG_wrapper.v"
  },
```
```
  {
    "class":"firrtl.transforms.BlackBoxInlineAnno",
    "target":"TestHarness.EICG_wrapper",
    "name":"EICG_wrapper.v",
    "text":"..."
  },
```